Release 0.13.30 (#3194)
digital-asset/daml: docs/source/support/release-notes.rst

This page contains release notes for the SDK.

.. _release-0-13-30:

0.13.30 - 2019-10-15
--------------------

DAML Standard Library
~~~~~~~~~~~~~~~~~~~~~

- Add ``DA.Action.State`` module containing a ``State`` action that
  can be used for computations that modify a state variable.

DAML Compiler
~~~~~~~~~~~~~

- Fixed the location of interface files when the
  ``source`` field in ``daml.yaml`` points to a file. This is mainly
  important for when you want to use the created ``.dar`` in the
  ``dependencies`` field of another package.
  See `issue #3135 <https://github.com/digital-asset/daml/issues/3135>`_.

DAML-LF
~~~~~~~

> - **Breaking** archive proto package renamed from `com.digitalasset.daml_lf` to `com.digitalasset.daml_lf_dev`

Reviewer: I wonder if we should tone this down.
```
Rename DAML-LF Archive protobuf package from `com.digitalasset.daml_lf` to `com.digitalasset.daml_lf_dev`. This will only affect you do not use the DAML-LF Archive reader provided with the SDK but a custom one based on code generation by protoc.

Author: :+1:

- **Breaking** Some bintray/maven packages are renamed:
   + `com.digitalasset.daml-lf-proto` becomes `com.digitalasset.daml-lf-dev-archive-proto`
   + `com.digitalasset.daml-lf-archive` becomes `com.digitalasset:daml-lf-dev-archive-java-proto`
   + `com.digitalasset.daml-lf-archive-scala` becomes `com.digitalasset.daml-lf-archive-reader`

- Add immutable bintray/maven packages for handling DAML-LF archive up to version 1.6 in a stable way:
   + `com.digitalasset.daml-lf-1.6-archive-proto`

     This package contains the archive protobuf definitions as they

>      were introduced when 1.6 was freeze.  Those definitions can be

Reviewer: ```suggestion
     were introduced when 1.6 was frozen.  These definitions can be
```

>      used to read DAML-LF archive up to version 1.6.

Reviewer: ```suggestion
     used to read DAML-LF archives up to version 1.6.
```

     The main advantage of this package over the `dev` version

>      (`com.digitalasset.daml-lf-dev-archive-proto`) is its

Reviewer: ```suggestion
     (`com.digitalasset.daml-lf-dev-archive-proto`) is that it is
```

>      immutability (it is guarantee it will never changed once

Reviewer: ```suggestion
     immutable (it is guaranteed to never changed once
```

>      introduced in the SDK) in other words one can used it without

Reviewer: ```suggestion
     introduced in the SDK). In other words, it can be used without
```

>      suffering frequent breaking change introduced in the `dev`

Reviewer: ```suggestion
     suffering from frequent breaking changes introduced in the `dev`
```

     version.

>      Going forward the SKD will contains a similar immutable package

Reviewer: ```suggestion
     Going forward the SKD will contain a similar immutable package
```

     containning the proto definition for at least each DAML-LF

>      version the compiler support.

Reviewer: ```suggestion
     version the compiler supports.
```

>      We strongly advice anyone reading daml-lf archive directly to use

Reviewer: ```suggestion
     We strongly advise anyone reading DAML-LF Archives directly to use
```

>      this package (or the under

Reviewer: ```suggestion
     this package (or the one under
```

>      `com.digitalasset:daml-lf-1.6-archive-java-proto` package), as

Reviewer: ```suggestion
     `com.digitalasset:daml-lf-1.6-archive-java-proto` package).
```

>      frequent breaking changes may be introduced in the `dev` without

Reviewer: ```suggestion
     Breaking changes to the `dev` version may be introduced frequently and without
```
